src/Utils.ps1

```diff
@@ -280,21 +280,37 @@ function Get-PathStringComparison {
 
 function Get-PromptPath {
     $settings = $global:GitPromptSettings
-    $abbrevHomeDir = $settings -and $settings.DefaultPromptAbbreviateHomeDirectory
+    $stringComparison = Get-PathStringComparison
 
     # A UNC path has no drive so it's better to use the ProviderPath e.g. "\\server\share".
     # However for any path with a drive defined, it's better to use the Path property.
     # In this case, ProviderPath is "\LocalMachine\My"" whereas Path is "Cert:\LocalMachine\My".
     # The latter is more desirable.
     $pathInfo = $ExecutionContext.SessionState.Path.CurrentLocation
     $currentPath = if ($pathInfo.Drive) { $pathInfo.Path } else { $pathInfo.ProviderPath }
+    if (!$settings -or !$currentPath -or $currentPath.Equals($Home, $stringComparison)) {
+        return $currentPath
+    }
 
-    $stringComparison = Get-PathStringComparison
+    $abbrevHomeDir = $settings.DefaultPromptAbbreviateHomeDirectory
+    $abbrevGitDir = $settings.DefaultPromptAbbreviateGitDirectory
 
-    # Abbreviate path by replacing beginning of path with ~ *iff* the path is under the user's home dir
-    if ($abbrevHomeDir -and $currentPath -and !$currentPath.Equals($Home, $stringComparison) -and
-        $currentPath.StartsWith($Home, $stringComparison)) {
+    # Look up the git root
+    if ($abbrevGitDir) {
+        $gitPath = Get-GitDirectory
+        # Up one level from `.git`
+        if ($gitPath) { $gitPath = Split-Path $gitPath -Parent }
```

Comment:
Two edge cases to consider here:

1. If the current repo is bare, this returns its parent directory.
2. `Get-GitDirectory` inside a worktree currently returns the path to `.git/worktrees/[name]`, which won't match the current path. I can think of two ways to resolve this:
    1. Revise `Get-GitDirectory` to return the contents of `.git/worktrees/[name]/gitdir`. No idea what the downstream effects of this would be.
    2. Check here if `$gitPath` matches `.git/worktrees/`; if it does, set `$gitPath` to the contents of `$gitPath/gitdir`

Reply:
Never used worktrees, but git abbreviation just won't happen in that case.
An issue should be created to track it.

Reply:
> Never used worktrees, but git abbreviation just won't happen in that case.

Not a problem. It doesn't break, it just won't work as expected in all cases.

> An issue should be created to track it.

#746